pkg/util/pod_utils.go

```diff
@@ -57,3 +64,94 @@ func IsConsistentWithRevision(pod *v1.Pod, revision string) bool {
 	}
 	return false
 }
+
+func FilterActivePods(pods []*v1.Pod) []*v1.Pod {
+	var activePods []*v1.Pod
+	for _, pod := range pods {
+		if pod.DeletionTimestamp.IsZero() {
+			activePods = append(activePods, pod)
+		}
+	}
+	return activePods
+}
+
+func ListOwnedPods(c client.Client, object client.Object) ([]*v1.Pod, error) {
+	selector, err := ParseSelector(object)
+	if err != nil {
+		return nil, err
+	}
+
+	podLister := &v1.PodList{}
+	err = c.List(context.TODO(), podLister, &client.ListOptions{LabelSelector: selector}, utilclient.DisableDeepCopy)
+	if err != nil {
+		return nil, err
+	}
+	pods := make([]*v1.Pod, 0, len(podLister.Items))
+	for i := range podLister.Items {
+		pod := &podLister.Items[i]
```

Comment:
we can safely filter out pods with p.Status.Phase = PodSucceeded and PodFailed

Reply:
done